src/web/components/SignInGate/SignInGateSelector.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useState, useEffect } from 'react';
+import { useAB } from '@guardian/ab-react';
+import { ABTest, Runnable } from '@guardian/ab-core';
+import { constructQuery } from '@root/src/lib/querystring';
+
+import { setUserDismissedGate } from '@frontend/web/components/SignInGate/dismissGate';
+import {
+    SignInGateComponent,
+    CurrentABTest,
+} from '@frontend/web/components/SignInGate/gateDesigns/types';
+import { getCookie } from '@frontend/web/browser/cookie';
+
+// Sign in Gate A/B Tests
+import { signInGatePatientia } from '@frontend/web/experiments/tests/sign-in-gate-patientia';
+import { signInGateMainVariant } from '@root/src/web/experiments/tests/sign-in-gate-main-variant';
+import { signInGateMainControl } from '@root/src/web/experiments/tests/sign-in-gate-main-control';
+
+// Sign in Gate Types
+import { signInGateComponent as gateMainVariant } from '@root/src/web/components/SignInGate/gates/main-variant';
+import { signInGateComponent as gateMainControl } from '@root/src/web/components/SignInGate/gates/main-control';
+import { signInGateComponent as gatePatientiaControl } from '@root/src/web/components/SignInGate/gates/patientia-control';
+import { signInGateComponent as gatePatientiaVariant } from '@root/src/web/components/SignInGate/gates/patientia-variant';
+
+import {
+    ComponentEventParams,
+    submitViewEventTracking,
+    withComponentId,
+} from './componentEventTracking';
+
+// component name, should always be sign-in-gate
+export const componentName = 'sign-in-gate';
+
+// interface for the sign in gate selector component
+interface SignInGateSelectorProps {
+    isSignedIn?: boolean;
+    CAPI: CAPIBrowserType;
+}
+
+// interface for the component which shows the sign in gate
+interface ShowSignInGateProps {
+    setShowGate: React.Dispatch<React.SetStateAction<boolean>>;
+    abTest: CurrentABTest;
+    CAPI: CAPIBrowserType;
+    signInUrl: string;
+    gateVariant: SignInGateComponent;
+}
+
+const dismissGate = (
+    setShowGate: React.Dispatch<React.SetStateAction<boolean>>,
+    currentAbTestValue: CurrentABTest,
+) => {
+    setShowGate(false);
+    setUserDismissedGate(currentAbTestValue.variant, currentAbTestValue.name);
+
+    // When the user closes the sign in gate, we scroll them back to the main content
+    const articleBody =
+        document.querySelector('.js-article__body') ||
+        document.querySelector('.article-body-commercial-selector');
+
+    articleBody?.parentElement?.scrollIntoView(true);
```

Comment:
We tried implementing the `useLayoutHook` but didn't scroll to the correct location on the article. It scrolled about 1/2 way down the article, rather than to the top of the article.

Reply:
The way it's currently implemented does scroll correctly to the top of the article.